src/main/java/org/openhab/automation/jrule/internal/items/JRulePersistenceExtensions.java

```diff
@@ -32,6 +39,46 @@
  * @author Arne Seime - Initial contribution
  */
 class JRulePersistenceExtensions {
+    public static void persist(String itemName, ZonedDateTime timestamp, JRuleValue state, String serviceId) {
+        Item item = getItem(itemName);
+        JRuleEngine jRuleEngine = JRuleEngine.get();
+        PersistenceServiceRegistry persistenceServiceRegistry = jRuleEngine.getPersistenceServiceRegistry();
+        PersistenceService persistenceService = persistenceServiceRegistry
+                .get(Optional.ofNullable(serviceId).orElse(persistenceServiceRegistry.getDefaultId()));
+        if (persistenceService instanceof ModifiablePersistenceService modifiablePersistenceService) {
+            modifiablePersistenceService.store(item, timestamp, state.toOhState());
+        } else {
+            throw new JRuleRuntimeException("cannot persist item state, persistence service (%s) not modifiable"
+                    .formatted(persistenceService.getId()));
+        }
+    }
+
+    public static Optional<JRuleValue> stateAt(String itemName, ZonedDateTime timestamp, String serviceId) {
+        Item item = getItem(itemName);
+        JRuleEngine jRuleEngine = JRuleEngine.get();
+        PersistenceServiceRegistry persistenceServiceRegistry = jRuleEngine.getPersistenceServiceRegistry();
+        PersistenceService persistenceService = persistenceServiceRegistry
+                .get(Optional.ofNullable(serviceId).orElse(persistenceServiceRegistry.getDefaultId()));
```

Comment:
Should we get default service here? Maybe throw error?

Reply:
While querying data it must always be a QueryablePersistenceService. In my eyes it's handled as the other query methods.